pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -220,6 +222,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		return nil
 	}
 
+	var defaults []v1alpha1.PipelineParam
+	if len(p.Spec.Params) > 0 {
+		defaults = append(defaults, p.Spec.Params...)
+	}
+
+	// Apply parameter templating from the PipelineRun
+	p = resources.ApplyParameters(p, pr, defaults...)
```

Comment:
`ApplyParameters ` function has access to params through `p` so why does the code pass params again as a parameter?

Reply:
Lemme think on this - I need to dive through the `taskrun` approach to understand it better.

Reply:
ah, right, `taskrun`'s equivalent needs this because it isn't getting passed the `TaskSpec`. We could switch to pass that too, but it feels simpler to just leave that alone.

Here, on the other hand, yeah, `defaults` is utterly pointless. Fixing. =)